modules/docs/arrow-docs/docs/docs/datatypes/validated/README.md

```diff
@@ -214,6 +214,79 @@ val houseNumber = config.parse(Read.intRead, "house_number").withEither { either
 houseNumber
 ```
 
+## Abstracting aways validation strategies with `ApplicativeError`
+
+We may use `ApplicativeError` instead of `Validated` to abstract away validation strategies and raising errors in the context we are computing in.
+
+```kotlin:ank:silent
+import arrow.*
+import arrow.core.*
+import arrow.typeclasses.*
+import arrow.data.*
+
+sealed class ValidationError(val msg: String)
+data class DoesNotContain(val value: String) : ValidationError("Did not contain $value")
+data class MaxLength(val value: Int) : ValidationError("Exceeded length of $value")
+data class NotAnEmail(val reasons: Nel<ValidationError>) : ValidationError("Not a valid email")
+
+data class FormField(val label: String, val value: String)
+data class Email(val value: String)
+
+abstract class Rules<F>(val A: ApplicativeError<F, Nel<ValidationError>>) : ApplicativeError<F, Nel<ValidationError>> by A {
```

Comment:
A doesn't have to be a field. Why the abstract class?

Reply:
There is nothing to implement, therefore no need for an interface and if it were an `interface` then it results in more boilerplate in the interpreters.
```kotlin
interface Rules<F> : ApplicativeError<F, Nel<ValidationError>> {

  val A: ApplicativeError<F, Nel<ValidationError>>

  private fun FormField.contains(needle: String): Kind<F, FormField> =
    if (value.contains(needle, false)) just(this)
    else raiseError(DoesNotContain(needle).nel())

  private fun FormField.maxLength(maxLength: Int): Kind<F, FormField> =
    if (value.length <= maxLength) just(this)
    else raiseError(MaxLength(maxLength).nel())

  fun FormField.validateEmail(): Kind<F, Email> =
    map(contains("@"), maxLength(250), {
      Email(value)
    }).handleErrorWith { raiseError(NotAnEmail(it).nel()) }

}

val errorAccumulatingApplicative : ApplicativeError<ValidatedPartialOf<Nel<ValidationError>>, Nel<ValidationError>> =
  Validated.applicativeError(NonEmptyList.semigroup())

object ErrorAccumulationStrategy : 
  Rules<ValidatedPartialOf<Nel<ValidationError>>>, 
  ApplicativeError<ValidatedPartialOf<Nel<ValidationError>>, Nel<ValidationError>> by errorAccumulatingApplicative {
    override val A: ApplicativeError<ValidatedPartialOf<Nel<ValidationError>>, Nel<ValidationError>>
      get() = Validated.applicativeError(NonEmptyList.semigroup())
}
```

Normally in tagless the practice I see is that if you can implement the algebra entirely with a type class and not leave any methods abstract to the interpreters then you don't need to use an interface or trait since the very definition of the algebra is abstract by relying on the type classes.